src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -536,6 +547,70 @@ AppendShardIdToConstraintName(AlterTableCmd *command, uint64 shardId)
 }
 
 
+static bool
+ExtendRowReferencesWalker(Node *node, RowRefWalkerState *state)
+{
+	if (node == NULL)
+	{
+		return false;
+	}
+
+	if (IsA(node, IndexElem))
+	{
+		IndexElem *indexElem = (IndexElem *) node;
+
+		return raw_expression_tree_walker(indexElem->expr, ExtendRowReferencesWalker,
+										  state);
+	}
+	else if (IsA(node, ColumnRef))
+	{
+		ColumnRef *columnRef = (ColumnRef *) node;
+		ListCell *fieldsCell;
+
+		/*
+		 * Append the shardId to any ColumnRef String values that are
+		 * equal to the relationName.  These are actually ROW(relname)
+		 * references.
+		 */
+		foreach(fieldsCell, columnRef->fields)
```

Comment:
On 2016-08-31 12:20:05 -0700, Eric Ridge wrote:
> > +		IndexElem *indexElem = (IndexElem *) node;
> > +
> > +		return raw_expression_tree_walker(indexElem->expr, ExtendRowReferencesWalker,
> > +										  state);
> > +	}
> > +	else if (IsA(node, ColumnRef))
> > +	{
> > +		ColumnRef *columnRef = (ColumnRef *) node;
> > +		ListCell *fieldsCell;
> > +
> > +		/*
> > +		 * Append the shardId to any ColumnRef String values that are
> > +		 * equal to the relationName.  These are actually ROW(relname)
> > +		 * references.
> > +		 */
> > +		foreach(fieldsCell, columnRef->fields)
> 
> I guess I don't fully understand why `zdb(wikipedia_changes)` is written as a single-Value list.  Is that because the statement hasn't been run through some additional phase of analysis/rewriting yet, or is this just a case that has to be accounted for separately?

It's not yet been through parse-analysis. The raw parser doesn't access
any catalogs etc. And we can't, at that point, send it through
parse-analyis, because it'd error out because of a non-existing
table. In the long run this work should be moved to the master instead,
to avoid this rigamarole.